internal/stack/serverless.go

```diff
@@ -0,0 +1,462 @@
+
+	project, err = sp.createClients(project)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create project client")
+	}
+
+	fleetURL := config.Parameters[paramServerlessFleetURL]
+	if true {
+		fleetURL, err = project.DefaultFleetServerURL()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get fleet URL: %w", err)
+		}
+	}
+	project.Endpoints.Fleet = fleetURL
+
+	return project, nil
+}
+
+func getProjectSettings(options Options) (projectSettings, error) {
+	s := projectSettings{
+		Name:         createProjectName(options),
+		Type:         options.Profile.Config(configProjectType, defaultProjectType),
+		Region:       options.Profile.Config(configRegion, defaultRegion),
+		StackVersion: options.StackVersion,
+	}
+
+	return s, nil
+}
+
+func createProjectName(options Options) string {
+	return fmt.Sprintf("elastic-package-test-%s", options.Profile.ProfileName)
+}
+
+func newServerlessProvider(profile *profile.Profile) (*serverlessProvider, error) {
+	client, err := serverless.NewClient()
+	if err != nil {
+		return nil, fmt.Errorf("can't create serverless provider: %w", err)
+	}
+
+	return &serverlessProvider{profile, client}, nil
+}
+
+func (sp *serverlessProvider) BootUp(options Options) error {
+	logger.Warn("Elastic Serverless provider is in technical preview")
+
+	config, err := LoadConfig(sp.profile)
+	if err != nil {
+		return fmt.Errorf("failed to load configuration: %w", err)
+	}
+
+	settings, err := getProjectSettings(options)
+	if err != nil {
+		return err
+	}
+
+	if settings.Type == "elasticsearch" {
+		return fmt.Errorf("serverless project type not supported: %s", settings.Type)
+	}
+
+	var project *serverless.Project
+
+	project, err = sp.currentProject(config)
+	switch err {
+	default:
+		return err
+	case errProjectNotExist:
+		logger.Infof("Creating %s project: %q", settings.Type, settings.Name)
+		config, err = sp.createProject(settings, options, config)
+		if err != nil {
+			return fmt.Errorf("failed to create deployment: %w", err)
+		}
+
+		project, err = sp.currentProject(config)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve latest project created: %w", err)
+		}
+
+		logger.Infof("Creating agent policy")
+		err = project.CreateAgentPolicy(options.StackVersion)
+		if err != nil {
+			return fmt.Errorf("failed to create agent policy: %w", err)
+		}
+
+		// logger.Infof("Replacing GeoIP databases")
+		// err = cp.replaceGeoIPDatabases(config, options, settings.TemplateID, settings.Region, payload.Resources.Elasticsearch[0].Plan.ClusterTopology)
+		// if err != nil {
+		// 	return fmt.Errorf("failed to replace GeoIP databases: %w", err)
+		// }
+	case nil:
+		logger.Debugf("%s project existed: %s", project.Type, project.Name)
+		printUserConfig(options.Printer, config)
+		// logger.Infof("Updating project %s", project.Name)
+		// err = sp.updateDeployment(project, settings)
+		// if err != nil {
+		// 	return fmt.Errorf("failed to update deployment: %w", err)
+		// }
```

Comment:
Leftover?

Reply:
Currently, I don't find how to update a project in Serverless.
Maybe it can be updated just the local agent container ?

Reply:
For now, I'll remove these lines